### DISCUSSION TITLE
2.0: Mapped[enum.Enum]

### DISCUSSION THREAD

**Discussion by Dreamsorcerer** [sqlalchemy/sqlalchemy#8868] [Usage Questions]
Should it be possible to use `enum.Enum`?

e.g.
```
class Media(enum.Enum):
    PHYSICAL = "Physical"
    DIGITAL = "Digital"

class MyModel(Base):
    ...
    media: Mapped[Media]
```

Currently this gives `sqlalchemy.exc.ArgumentError: Could not locate SQLAlchemy Core type for Python type: <enum 'Media'>`

Or do we need to use `sqlalchemy.Enum` for this?

**Comment by Dreamsorcerer** [comment]
Also worth noting there are no typing errors. Don't know if it's possible to restrict `Mapped[]` allowed types to only valid types or not?

**Comment by CaselIT** [comment]
This is a duplicate of https://github.com/sqlalchemy/sqlalchemy/discussions/8856, please continue there
